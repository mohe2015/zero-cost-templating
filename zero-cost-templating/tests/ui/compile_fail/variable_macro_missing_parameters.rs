#![feature(coroutines)]

extern crate alloc;

use alloc::borrow::Cow;

use zero_cost_templating_macros::template_stream;

#[template_stream("test.html.hbs")]
pub async fn test() {
    let template = test_initial0!();
    let template = test_template0!(template);
    let page_title = Cow::from("thetitle");
    let template = test_page_title1!();
    let csrf_token = Cow::from("thetoken");
    let mut template = test_csrf_token2!(template, csrf_token);
    let template = test_template6!(template);
    test_copyright_year7!(template, Cow::from("2023".to_string()));
}

pub fn main() {}
