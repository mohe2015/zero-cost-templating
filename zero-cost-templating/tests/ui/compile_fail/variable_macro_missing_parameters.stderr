error: unexpected end of macro invocation
  --> tests/ui/compile_fail/variable_macro_missing_parameters.rs:14:20
   |
9  | #[template_stream("test.html.hbs")]
   | ----------------------------------- when calling this macro
...
14 |     let template = test_page_title1!();
   |                    ^^^^^^^^^^^^^^^^^^^ missing tokens in macro arguments
   |
note: while trying to match meta-variable `$template:expr`
  --> tests/ui/compile_fail/variable_macro_missing_parameters.rs:9:1
   |
9  | #[template_stream("test.html.hbs")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in the attribute macro `template_stream` (in Nightly builds, run with -Z macro-backtrace for more info)
